base/channels.jl

```diff
@@ -26,10 +26,21 @@ type Channel{T} <: AbstractChannel
         end
         new(Condition(), Condition(), :open, Array{T}(0), sz, Array{Condition}(0))
     end
+
+    # deprecated empty constructor
+    function Channel()
+        depwarn(string("The empty constructor Channel() is deprecated. ",
```

Comment:
The `Channel{T}()` constructor needs to be inside the type definition, while `Channel()` which is equivalent of `Channel{Any}()` can be outside. Thus added both in channels.jl itself.